moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -53,6 +53,26 @@ RobotTrajectory::RobotTrajectory(const robot_model::RobotModelConstPtr& robot_mo
 {
 }
 
+void RobotTrajectory::copy(const RobotTrajectory& robot_traj)
+{
+  this->robot_model_ = robot_traj.robot_model_;
+  this->group_ = robot_traj.group_;
+  this->waypoints_ = robot_traj.waypoints_;
+  this->duration_from_previous_ = robot_traj.duration_from_previous_;
+}
+
+void RobotTrajectory::deepCopy(const RobotTrajectory& robot_traj)
+{
+  this->robot_model_ = robot_traj.robot_model_;
```

Comment:
Is this really a deep copy of the RobotModel?

Reply:
Thanks for the review! No, this isn't a deep copy. But I observer that constructor is receiving a shared ptr so I thought I should assigned it to a same shared pointer. But now I get the picture, value should be same not the memory! Sorry, I am not much familiar with this part of the package.

Reply:
i believe we actually want to keep the same `RobotModel` in the copied trajectory.

the "deep" is supposed to imply that RobotStates are copied, not the whole underlying moveit class structure.

other proposals for this name are welcome.

Reply:
> i believe we actually want to keep the same `RobotModel` in the copied trajectory.

I agree

> the "deep" is supposed to imply that RobotStates are copied, not the whole underlying moveit class structure.
> 
> other proposals for this name are welcome.

I think "deep" could be a bit misleading. I would prefer something more explicit, maybe `copyWithRobotStates()`.

Reply:
Perhaps `RobotTrajectory::copy(const RobotTrajectory& other, bool copy_robot_states = false)` would be more explicit. This is the approach used in `robot_state/conversions.h`